src/Performance.php

```diff
@@ -0,0 +1,214 @@
+      echo '<link rel="preload" href="', esc_attr($source), '" as="style" onload="this.onload=null;this.rel=\'stylesheet\'">', "\n";
+      echo '<noscript><link rel="stylesheet" href="', esc_attr($source), '"></noscript>', "\n";
+      wp_dequeue_style($handle);
+    }
+  }
+
+  /**
+   * Gets the source for a given script or style handle.
+   *
+   * @return string
+   *   Script or style source.
+   */
+  public static function getHandleSource(\_WP_Dependency $handle): string {
+    // If the version parameter is a Boolean value the WP version is appendend
+    // to the script, nothing otherwise.
+    if ($handle->ver === NULL) {
+      $version = '';
+    }
+    else {
+      $version = '?ver=' . ($handle->ver ? $handle->ver : Plugin::$version);
+    }
+    $source = $handle->src . $version;
+    return $source;
+  }
+
+  /**
+   * Loads scripts as deferred or async.
+   *
+   * @implements script_loader_tag
+   */
+  public static function script_loader_tag($tag, $handle) {
+    $scripts_load = static::getAsyncScripts();
+
+    if (isset($scripts_load[$handle]) && strpos($tag, ' defer ') === FALSE && strpos($tag, ' async ') === FALSE) {
+      $tag = str_replace(' src=', sprintf(' %s src=', $scripts_load[$handle]), $tag);
+    }
+
+    return $tag;
+  }
+
+  /**
+   * Prefetches DNS entries for particular resources.
+   *
+   * @implements wp_resource_hints
+   */
+  public static function wp_resource_hints($urls, $relation_type) {
+    if ($relation_type === 'dns-prefetch') {
+      $urls = array_merge($urls, static::getResourcesUrlPrefetch());
+    }
+    return $urls;
+  }
+
+  /**
+   * Gets URLs of resources to prefetch.
+   *
+   * @return array
+   *   URLs of resources to prefetch.
+   */
+  public static function getResourcesUrlPrefetch() {
+    return apply_filters(Plugin::L10N . '/esources_url_prefetch', static::RESOURCES_URL_PREFETCH);
+  }
+
+  /**
+   * Gets scripts handles to load async.
+   *
+   * @return array
+   *   Handles of scripts to async loaded.
+   */
+  public static function getAsyncScripts(): array {
+    return apply_filters(Plugin::L10N . '/scripts_async_load', static::SCRIPTS_ASYNC_LOAD);
+  }
+
+  /**
+   * Gets styles handles to load async.
+   *
+   * @return array
+   *   Handles of styles to async loaded.
+   */
+  public static function getAsyncStyles(): array {
+    return apply_filters(Plugin::L10N . '/styles_async_load', static::SCRIPTS_ASYNC_LOAD);
+  }
+
+  /**
+   * Gets scripts handles to dequeue.
+   *
+   * @return array
+   *   Handles of scripts to be dequeued.
+   */
+  public static function getDequeueScripts(): array {
+    return apply_filters(Plugin::L10N . '/scripts_dequeue', static::SCRIPTS_DEQUEUE);
+  }
+
+  /**
+   * Gets styles handles to dequeue.
+   *
+   * @return array
+   *   Handles of styles to be dequeued.
+   */
+  public static function getDequeueStyles(): array {
+    return apply_filters(Plugin::L10N . '/styles_dequeue', static::STYLES_DEQUEUE);
```

Comment:
The filter should match the method name (and elsewhere). IMO it's more clear if you put the verb before the noun (`dequeue_styles`, `async_load_styles`, `async_load_scripts`). Another option would be to group scripts and styles together 

```
styles_async
styles_dequeue
```

```suggestion
    return apply_filters(Plugin::L10N . '/dequeue_styles', static::DEQUEUE_STYLES);
```